src/propcache/_helpers_py.py

```diff
@@ -46,13 +46,13 @@ def __init__(self, wrapped: Callable[..., _T]) -> None:
         self.name = wrapped.__name__
 
     @overload
-    def __get__(self, inst: None, owner: Optional[Type[Any]] = None) -> Self: ...
+    def __get__(self, inst: None, owner: Optional[Type[object]] = None) -> Self: ...
 
     @overload
-    def __get__(self, inst: _TSelf[_T], owner: Optional[Type[Any]] = None) -> _T: ...
+    def __get__(self, inst: _TSelf[_T], owner: Optional[Type[object]] = None) -> _T: ...
 
     def __get__(
-        self, inst: Optional[_TSelf[_T]], owner: Optional[Type[Any]] = None
+        self, inst: Optional[_TSelf[_T]], owner: Optional[Type[object]] = None
```

Comment:
Interestingly, the example @ https://docs.python.org/3/library/typing.html#typing.overload shows the `@overload` definitions as typed but the implementation not having any typing signature. Can it be that we need to move this to another overload?

Reply:
I've highlighted that before. Unfortunately, not typing the implementation leads to mypy not type checking it. It should really type check the implementation against each overload (as it does with TypeVar), but currently will only type check with the implementation annotations, which leads to a lot of awkward annotations or type ignores to get it passing in many instances.

Reply:
This is why we end up with a Union here. But, calling code would not know the correct return type from the union, it gets it from the overloads instead, which are more precise.

Reply:
Oh, interesting..